tlslite/tlsconnection.py

```diff
@@ -1532,13 +1532,24 @@ def _serverGetClientHello(self, settings, certChain, verifierDB,
 
         # Sanity check the SNI extension
         sniExt = clientHello.getExtension(ExtensionType.server_name)
+        # check if extension is well formed
+        if sniExt and (not sniExt.extData or not sniExt.serverNames):
+            for result in self._sendError(
+                    AlertDescription.decode_error,
+                    "Recevived SNI extension is malformed"):
+                yield result
         if sniExt and sniExt.hostNames:
             # RFC 6066 limitation
             if len(sniExt.hostNames) > 1:
                 for result in self._sendError(
                         AlertDescription.illegal_parameter,
                         "Client sent multiple host names in SNI extension"):
                     yield result
+            if not sniExt.hostNames[0]:
+                for result in self._sendError(
+                        AlertDescription.decode_error,
+                        "Received SNI extension is malformed"):
+                    yield result
```

Comment:
That's an empty hostname, right? In gnutls I send the ILLEGAL_PARAMETER parameter alert as I saw it more appropriate.